pyriemann/utils/mean.py

```diff
@@ -733,3 +734,144 @@ def nanmean_riemann(covmats, tol=10e-9, maxiter=100, init=None,
         sample_weight=sample_weight
     )
     return C
+
+
+###############################################################################
+
+
+def median_euclid(covmats, tol=10e-6, maxiter=50, init=None):
+    """Median of matrices according to the Euclidean metric.
+
+    Return the Euclidean geometric median for matrices,
+    different from the marginal median provided by numpy.median().
+
+    Parameters
+    ----------
+    covmats : ndarray, shape (n_matrices, n_channels, n_channels)
+        Set of matrices.
+    tol : float, default=10e-6
+        The tolerance to stop the gradient descent.
```

Comment:
it's not really a gradient descent you do here don't you? to me it's more like iterative reweighted least square

Reply:
From Section 5 of [Fletcher2009 ](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2735114/):
> The procedure iteratively updates the estimate mk of the geometric median using essentially a steepest descent on the weighted sum-of-distances function